node-build.sh

```diff
@@ -7,6 +7,13 @@ shopt -s nullglob
 
 DIST="./dist"
 
+# show input environment args
+echo "env: BITBUCKET_CLONE_DIR=$BITBUCKET_CLONE_DIR"
```

Comment:
These environment variables should never be echoed as they contain secrets.

Reply:
As does printenv, maybe add a verbose option to this script for debugging purposes?

Reply:
The reason I'm against echoing environment variables is that containers are usually handled way more securely than their output (e.g STDOut/err logs).

Just take a look at any ci/cd pipeline. the STDOut is always saved for way longer than the executing environment itself.

To me, it offers no benefit to include this code in the script at all. What do you hope to gain from it? Maybe I'm missing something.

Reply:
Just for debugging

Reply:
I would remove the access token (as it is both read/write) and wrap it in the verbose check that you have on the node container. That seems fair.

(and yes, I do hate that bitbucket doesn't keep 'secure' ENV variable secret in logs/stdout)